.changesets/fix_arc_config_mem_leak.md

```diff
@@ -0,0 +1,5 @@
+### Fix memory leak caused by Arc circular reference in `Notify` ([PR #3692](https://github.com/apollographql/router/pull/3692))
```

Comment:
```suggestion
### Fix memory leak caused by Arc circular reference in `Notify` ([Issue #3686](https://github.com/apollographql/router/issues/3686))
```

Reply:
Mostly changing in this PR just to trigger CI as me, since it was failing for unclear CircleCI auth/config reasons.  Thanks for the changelog! 😄 😍